src/main/java/org/threadly/concurrent/UnfairExecutor.java

```diff
@@ -0,0 +1,267 @@
+      for (Worker w : schedulers) {
+        w.addTask(new ShutdownTask(w));
+      }
+    }
+  }
+
+  /**
+   * Stops any new tasks from being submitted to the pool.  If any tasks are waiting for execution 
+   * they will be prevented from being run.  If a task is currently running it will be allowed to 
+   * finish (though this call will not block waiting for it to finish).
+   * 
+   * @return returns a list of runnables which were waiting in the queue to be run at time of shutdown
+   */
+  public List<Runnable> shutdownNow() {
+    shutdownStarted.set(true);
+    
+    List<Runnable> result = new ArrayList<Runnable>();
+    for (Worker w : schedulers) {
+      w.stopIfRunning();
+      Iterator<Runnable> it = w.taskQueue.iterator();
+      while (it.hasNext()) {
+        Runnable task = it.next();
+        it.remove();
+        if (! (task instanceof ShutdownTask)) {
+          result.add(task);
+        }
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Block until the thread pool has shutdown and all threads have been stopped.  If neither 
+   * {@link #shutdown()} or {@link #shutdownNow()} is invoked, then this will block forever.
+   * 
+   * @throws InterruptedException Thrown if blocking thread is interrupted waiting for shutdown
+   */
+  public void awaitTermination() throws InterruptedException {
+    for (Worker w : schedulers) {
+      w.thread.join();
+    }
+  }
+
+  /**
+   * Block until the thread pool has shutdown and all threads have been stopped.  If neither 
+   * {@link #shutdown()} or {@link #shutdownNow()} is invoked, then this will block until the 
+   * timeout is reached.
+   * 
+   * @param timeoutMillis time to block and wait for thread pool to shutdown
+   * @return {@code true} if the pool has shutdown, false if timeout was reached
+   * @throws InterruptedException Thrown if blocking thread is interrupted waiting for shutdown
+   */
+  public boolean awaitTermination(long timeoutMillis) throws InterruptedException {
+    long startTime = Clock.accurateForwardProgressingMillis();
+    for (Worker w : schedulers) {
+      long remainingWait = timeoutMillis - (Clock.lastKnownForwardProgressingMillis() - startTime);
+      if (remainingWait <= 0) {
+        return false;
+      }
+      w.thread.join(remainingWait);
+      if (w.thread.isAlive()) {
+        return false;
+      }
+    }
+    return true;
+  }
+  
+  @Override
+  protected void finalize() throws Throwable {
+    shutdown();
+    super.finalize();
+  }
+  
+  /**
+   * <p>Worker task for executing tasks on the provided thread.  This worker maintains an internal 
+   * queue for which tasks can be added on.  It will park itself once idle, and resume if tasks 
+   * are later then added.</p>
+   *  
+   * @author jent - Mike Jensen
+   * @since 4.5.0
+   */
+  private static class Worker extends AbstractService implements Runnable {
+    protected final Thread thread;
+    private final Queue<Runnable> taskQueue;
```

Comment:
I created a few other structures, and they were faster in some situations.  CLQ seems to be a pretty good consistent performer in all situations I have thrown at it.